apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -7,37 +7,61 @@
    * This is PinAppNavigation's constructor.
    * @param {Object} list [DOM container that contains pin elements]
    */
-  function PinAppNavigation(list) {
-    if (!list && typeof list !== 'object') {
-      console.error('List of pin apps must be an object');
-      return;
+  function PinAppNavigation(list, selector) {
+    if (!list) {
+      console.error('List must be a dom container with navigated elements');
     }
 
-    this.elemList = list;
-    window.removeEventListener('keydown', this);
-    window.addEventListener('keydown', this);
-  }
+    /**
+     * This variable contains all elements that will get using this.selector
+     * @type {Array}
+     */
+    this.elements = [];
 
+    /**
+     * Index of element in this.elements that is selected
+     * @type {Number}
+     */
+    this.selectedElemIndex = 0;
 
-  PinAppNavigation.prototype = {
+    /**
+     * DOM container that has all elements
+     * @type {Dom container}
+     */
+    this.elemList = null;
 
-    selector: '',
-    elements: null,
-    preventDef: false,
-    selectedElemIndex: 0,
-    elemList: null,
-    prevVerticalKey: null,
+    /**
+     * This variable is used to know selected element
+     * has middle size or large size
+     * @type {Boolean}
+     */
+    this.middleElem = null;
+
+    this.elemList = list;
```

Comment:
Am I missing something, or you just assigned 'null' 9 lines earlier? :open_mouth: